_posts/2020-01-17-strong-parameters-migration-guide.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: "The Complete Guide to Migrate to Strong Parameters"
+date: 2020-01-21 13:00:00
+categories: ["rails", "upgrades"]
+author: luciano
+---
+
+Migrating from `Protected Attributes` to `Strong Parameters` in a [Rails](https://rubyonrails.org/) project can be a huge step of the upgrade process. Specially when we are [upgrading a large application](https://www.fastruby.io/blog/rails/upgrades/case-study/upgrading-a-large-rails-application-from-4.2-to-5.2.html). This guide is meant to help you tackle that step faster and with a lot less pain.
+
+<!--more-->
+
+### Protected Attributes & Strong Parameters
+
+To give you a bit of context, let's recap what `Protected Attributes` and [Strong Parameters](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters) actually are. They are two different Rails implementations for protecting attributes from end-user injection (a.k.a. [Mass Assignment](https://owasp.org/www-project-cheat-sheets/cheatsheets/Mass_Assignment_Cheat_Sheet.html))
+
+To understand what are the benefits of `Strong Parameters` over `Protected Attributes`, I recommend you checking [this RailsCasts episode](http://railscasts.com/episodes/371-strong-parameters?autoplay=true).
+
+`Protected Attributes` was part of the Rails core since the beginning of it until Rails 3.2. In Rails 4.0 they introduced `Strong Parameters` as a replacement of it, and it has been part of the core since then. Before that it was possible to use it through the [strong_parameters](https://github.com/rails/strong_parameters) gem.
```

Comment:
```suggestion
`Protected Attributes` were part of the Rails core since the beginning of it until Rails 3.2. In Rails 4.0 they introduced `Strong Parameters` as a replacement and they have been part of the core since then. Before that it was possible to use them through the [strong_parameters](https://github.com/rails/strong_parameters) gem.
```

Reply:
@ABizzinotto hm, not sure about this one. `Protected Attributes` is considered a feature (same with `Strong Parameters`), so I guess we can reference it as a singular?